src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -379,346 +64,97 @@
+      case EnumUtilities.isTypeAlteration(filing): return 'alteration-filing'
+      case EnumUtilities.isTypeChangeOfAddress(filing): return 'change-of-address'
+      case EnumUtilities.isTypeConsentContinuationOut(filing): return 'consent-continuation-out'
+      case EnumUtilities.isTypeDissolutionVoluntary(filing): return 'dissolution-voluntary'
+      case EnumUtilities.isTypeIncorporationApplication(filing): return 'incorporation-application'
+      case EnumUtilities.isTypeRestorationLimited(filing): return 'limited-restoration'
+      case EnumUtilities.isTypeRestorationConversion(filing): return 'limited-restoration-conversion'
+      case EnumUtilities.isTypeRestorationExtension(filing): return 'limited-restoration-extension'
+      case EnumUtilities.isTypeRegistration(filing): return 'registration-filing'
+      case EnumUtilities.isTypeStaff(filing): return 'staff-filing'
+      default: return 'default-filing'
     }
-
-    // report number of items back to parent (dashboard)
-    this.$emit('history-count', this.historyItems.length)
-
-    // Check if there is a pending (ie, paid / not yet completed) filing.
-    // This is a blocker because it needs to be completed first.
-    // Also update pending COA data.
-    let isCoaPending = false
-    let coaEffectiveDate: Date = null
-    const blockerFiling = this.historyItems.some(item => {
-      if (EnumUtilities.isStatusPaid(item)) {
-        if (item.isFutureEffectiveCoaPending) {
-          isCoaPending = true
-          coaEffectiveDate = item.effectiveDate
-        }
-        return true
-      }
-      return false
-    })
-    this.setIsCoaPending(isCoaPending)
-    this.setCoaEffectiveDate(coaEffectiveDate)
-    this.setHasBlockerFiling(blockerFiling)
-
-    // if needed, highlight a specific filing
-    if (this.highlightId) this.highlightFiling(this.highlightId)
   }
 
-  /** Loads a filing into the History Items array. */
-  private loadFiling (filing: ApiFilingIF): void {
-    try {
-      // NB: these `new Date()` are safe because the date strings are in GMT (ie, UTC)
-      //     so the conversion to JS Date ignores the browser's local timezone
-      const effectiveDate = new Date(filing.effectiveDate)
-      const submittedDate = new Date(filing.submittedDate)
-
-      // build filing item
-      const item: HistoryItemIF = {
-        availableOnPaperOnly: filing.availableOnPaperOnly,
-        commentsCount: filing.commentsCount,
-        displayName: filing.displayName,
-        effectiveDate,
-        filingId: filing.filingId,
-        isFutureEffective: filing.isFutureEffective,
-        name: filing.name || FilingTypes.UNKNOWN,
-        status: filing.status,
-        submittedDate,
-        submitter: filing.submitter,
-
-        commentsLink: filing.commentsLink,
-        documentsLink: filing.documentsLink,
-        filingLink: filing.filingLink,
-
-        comments: null, // null until loaded
-        documents: null // null until loaded
-      }
```

Comment:
No need to do this any more -- it's the same list fetched from the API (plus a couple of properties).